packages/status-js/src/client/chat.ts

```diff
@@ -0,0 +1,418 @@
+import { hexToBytes } from 'ethereum-cryptography/utils'
+import { PageDirection } from 'js-waku'
+
+import { ChatMessage as ChatMessageProto } from '~/protos/chat-message'
+import { CommunityRequestToJoin } from '~/protos/communities'
+import { EmojiReaction } from '~/protos/emoji-reaction'
+
+import { idToContentTopic } from '../contentTopic'
+import { createSymKeyFromPassword } from '../encryption'
+import { getReactions } from './community/get-reactions'
+
+import type { MessageType } from '../../protos/enums'
+import type { Client } from '../client'
+import type { Community } from './community/community'
+import type { Reactions } from './community/get-reactions'
+import type { ImageMessage } from '~/src/proto/communities/v1/chat_message'
+import type { CommunityChat } from '~/src/proto/communities/v1/communities'
+import type { WakuMessage } from 'js-waku'
+
+export type ChatMessage = ChatMessageProto & {
+  messageId: string
+  pinned: boolean
+  reactions: Reactions
+  chatUuid: string
+  responseToMessage?: Omit<ChatMessage, 'responseToMessage'>
+}
+
+// todo?: add clock
+export class Chat {
+  // todo?: add whole community reference
+  private readonly community: Community
```

Comment:
Notice: references whole Community to only use `this.community.publicKey` – at least for now